### PR TITLE
CPBR-2896 add sr-ready support to ub

### PR DESCRIPTION
**Pull request by hk10111**
### Change Description

Added a command to check if schema-registry is ready
Added a helper to make requests
Added tests with combinations around env vars and valid/invalid combinations of hosts/ports 
### Testing

Tested by building ub.go locally and running a sr docker container and checked the sr-ready command by providing the host and port of docker container and it worked

### CODE REVIEW

File: base-java/ub/ub.go [hunk 1]
```diff
@@ -437,6 +451,82 @@ func checkKafkaReady(minNumBroker string, timeout string, bootstrapServers strin
...
+// It supports TLS, certificate validation, and basic authentication.
+func makeRequest(host string, port int, secure bool, ignoreCert bool, username string, password string, path string) (*http.Response, error) {
+	scheme := "https"
+	if !secure {
+		scheme = "http"
+	}
+	
+	url := fmt.Sprintf("%s://%s:%d/%s", scheme, host, port, path)
+	
+	// Create HTTP client with timeout
```
Comment (rahejaprince) [inline base-java/ub/ub.go:464]: please remove redundant comments. If comments says what can be derived from code easily, those comments can be removed.

File: base-java/ub/ub.go [hunk 2]
```diff
@@ -437,6 +451,82 @@ func checkKafkaReady(minNumBroker string, timeout string, bootstrapServers strin
...
+	httpClient := &http.Client{
+		Timeout: 30 * time.Second, // Default timeout
```
Comment (rahejaprince) [inline base-java/ub/ub.go:466]: let's define the timeout as constant on the top.

File: base-java/ub/ub.go [hunk 3]
```diff
@@ -437,6 +451,82 @@ func checkKafkaReady(minNumBroker string, timeout string, bootstrapServers strin
...
+	}
+	
+	// Create request
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+	
+	// Add authentication if provided
+	if username != "" || password != "" {
```
Comment (rahejaprince) [inline base-java/ub/ub.go:485]: should this be `username != "" && password != ""`? In this logic, if any of the one is set might lead to authentication error. How is this implemented in cub?

File: base-java/ub/ub.go [hunk 4]
```diff
@@ -437,6 +451,82 @@ func checkKafkaReady(minNumBroker string, timeout string, bootstrapServers strin
...
+	
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading response body: %v\n", err)
+		return false
+	}
+	
+	// Check if response is successful and contains 'compatibilityLevel'
+	statusOK := resp.StatusCode >= 200 && resp.StatusCode < 300
```
Comment (rahejaprince) [inline base-java/ub/ub.go:521]: please use internal http library for status codes. You can refer https://go.dev/src/net/http/status.go

File: base-java/ub/ub.go [hunk 5]
```diff
@@ -588,6 +678,25 @@ func runKafkaReadyCmd(_ *cobra.Command, args []string) error {
 	return nil
 }
 
+func runSrReadyCmd(_ *cobra.Command, args []string) error {
```
Comment (rahejaprince) [inline base-java/ub/ub.go:681]: nit: rename the method name to `runSchemaReigstryReadyCmd` is easier to understand

File: base-java/ub/ub.go [hunk 6]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/spf13/cobra"
```
Comment (rahejaprince) [inline base-java/ub/ub.go:24]: Please add unit tests for this

File: base-java/ub/ub.go (hunk 1 above)
Comment (hk10111) [reply base-java/ub/ub.go]: done

File: base-java/ub/ub.go (hunk 6 above)
Comment (hk10111) [reply base-java/ub/ub.go:24]: added appropriate tests around env vars and optional flags like username, password

File: base-java/ub/ub.go [hunk 7]
```diff
@@ -437,6 +455,65 @@ func checkKafkaReady(minNumBroker string, timeout string, bootstrapServers strin
...
+	
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %w", err)
+	}
+	
+	statusOK := resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices
+	if statusOK && strings.Contains(string(body), "compatibilityLevel") {
+		return nil
+	} else {
```
Comment (rahejaprince) [inline base-java/ub/ub.go:512]: nit: this can be simplified to 
```
if ok {
  return nil
}
return err
```

File: base-java/ub/ub_test.go [hunk 8]
```diff
@@ -1176,3 +1180,212 @@ func Test_getEnvWithFallbacks(t *testing.T) {
...
+			envVars: map[string]string{},
+			args:    []string{"localhost", "8081", "5"},
+			expectError: false,
+		},
+		{
+			name: "with username and password",
+			envVars: map[string]string{},
+			args:    []string{"localhost", "8081", "5"},
+			expectError: false,
+		},
```
Comment (rahejaprince) [inline base-java/ub/ub_test.go:1346]: all the inputs seem to be same? Is this expected? We should different edge cases in the unit tests.
Also envVars is always empty in the test case.

**Review by rahejaprince — APPROVED** [review]
